src/rust/catcollar/mod.rs

```diff
@@ -433,7 +389,10 @@ impl CatcollarLibOS {
         // close was successful.
         match Self::do_close(fd, yielder).await {
             Ok(()) => {
-                qtable.borrow_mut().free(&qd);
+                qtable
+                    .borrow_mut()
+                    .free::<CatcollarQueue>(&qd)
+                    .expect("queue shouild exist");
```

Comment:
To follow out guidelines, could you add a comment here to explain why it is safe to call `expect()` here?